utilities/include/alps/numeric/tensors/tensor_base.hpp

```diff
@@ -316,6 +374,14 @@ namespace alps {
           return storage_.size();
         }
 
+        /**
+         * Negate tensor
+         */
+        tensor_base < T, Dim, Container > operator-() {
+          *this *= -1.0;
+          return *this;
+        };
```

Comment:
This is strange behaviour for a unary minus. It modifies `this` in-place and then returns a copy of it.

Usually, it is a `const` method and returns a new object containing `-a`.  In other words, `-a` should behave like `0-a`.

Once you're at it, there should be a test for this I think.